File: components/common/src/androidMain/kotlin/com/mirego/pilot/components/ui/type/PilotTextContentType.kt
package com.mirego.pilot.components.ui.type

import androidx.compose.ui.autofill.ContentType
import com.mirego.pilot.components.InternalPilotComponentsApi
import com.mirego.pilot.components.type.PilotTextContentType

@InternalPilotComponentsApi
public val PilotTextContentType.composeValue: ContentType?
    get() = when (this) {
        PilotTextContentType.NotSet -> null
        PilotTextContentType.Name -> ContentType.PersonFullName
        PilotTextContentType.NamePrefix -> ContentType.PersonNamePrefix
        PilotTextContentType.GivenName -> ContentType.PersonFirstName
        PilotTextContentType.MiddleName -> ContentType.PersonMiddleName
        PilotTextContentType.FamilyName -> ContentType.PersonLastName
        PilotTextContentType.NameSuffix -> ContentType.PersonNameSuffix
        PilotTextContentType.Nickname -> ContentType.PersonFullName
        PilotTextContentType.JobTitle -> null
        PilotTextContentType.OrganizationName -> null
        PilotTextContentType.Location -> null
        PilotTextContentType.FullStreetAddress -> ContentType.AddressStreet
        PilotTextContentType.StreetAddressLine1 -> ContentType.AddressStreet
        PilotTextContentType.StreetAddressLine2 -> ContentType.AddressAuxiliaryDetails
        PilotTextContentType.AddressCity -> ContentType.AddressLocality
        PilotTextContentType.AddressState -> ContentType.AddressRegion
        PilotTextContentType.AddressCityAndState -> ContentType.AddressLocality + ContentType.AddressRegion
        PilotTextContentType.Sublocality -> ContentType.AddressLocality
        PilotTextContentType.CountryName -> ContentType.AddressCountry
        PilotTextContentType.PostalCode -> ContentType.PostalCode
        PilotTextContentType.TelephoneNumber -> ContentType.PhoneNumber
        PilotTextContentType.EmailAddress -> ContentType.EmailAddress
        PilotTextContentType.URL -> null
        PilotTextContentType.CreditCardNumber -> ContentType.CreditCardNumber
        PilotTextContentType.Username -> ContentType.Username
        PilotTextContentType.Password -> ContentType.Password
        PilotTextContentType.NewPassword -> ContentType.NewPassword
        PilotTextContentType.OneTimeCode -> ContentType.SmsOtpCode
    }

File: components/ios/base/Types/PilotTextContentType.swift
import Shared
import SwiftUI

extension PilotTextContentType {
    public var uiTextContentType: UITextContentType? {
        switch self {
        case .notSet:
            return nil
        case .name:
            return .name
        case .namePrefix:
            return .namePrefix
        case .givenName:
            return .givenName
        case .middleName:
            return .middleName
        case .familyName:
            return .familyName
        case .nameSuffix:
            return .nameSuffix
        case .nickname:
            return .nickname
        case .jobTitle:
            return .jobTitle
        case .organizationName:
            return .organizationName
        case .location:
            return .location
        case .fullStreetAddress:
            return .fullStreetAddress
        case .streetAddressLine1:
            return .streetAddressLine1
        case .streetAddressLine2:
            return .streetAddressLine2
        case .addressCity:
            return .addressCity
        case .addressState:
            return .addressState
        case .addressCityAndState:
            return .addressCityAndState
        case .sublocality:
            return .sublocality
        case .countryName:
            return .countryName
        case .postalCode:
            return .postalCode
        case .telephoneNumber:
            return .telephoneNumber
        case .emailAddress:
            return .emailAddress
        case .url:
            return .URL
        case .creditCardNumber:
            return .creditCardNumber
        case .username:
            return .username
        case .password:
            return .password
        case .newPassword:
            return .newPassword
        case .oneTimeCode:
            return .oneTimeCode
        }
    }
}
